test/idp_fraud_data_handler_test.py

```diff
@@ -0,0 +1,750 @@
+import os
+import urllib.parse
+import uuid
+from unittest import TestCase
+
+import boto3
+import dateparser
+import psycopg2
+from moto import mock_s3, mock_kms
+from parameterized import parameterized
+from retrying import retry
+from testfixtures import LogCapture
+
+from src import idp_fraud_data_handler, database, event_mapper
+from src.database import RunInTransaction
+from src.idp_fraud_event import IdpFraudEvent
+from test.helpers import IDP_ENTITY_ID, clean_db, create_fraud_event_string, file_exists_in_s3, setup_stub_aws_config, \
+    DB_PASSWORD
+
+UPLOAD_BUCKET_NAME = 's3-idp-fraud-data-bucket'
+UPLOAD_FILE_NAME = 'idp-data.csv'
+UPLOAD_USERNAME = 'my.user.name@example.com'
+
+
+@mock_s3
+@mock_kms
+class IdpFraudDataHandlerTest(TestCase):
+    __s3_client = None
+    __kms_client = None
+    __queue_url = None
+    __key_id = None
+    db_connection = None
+    db_connection_string = "host='event-store' dbname='events' user='postgres'"
+
+    @classmethod
+    def setUpClass(cls):
+        cls.connect()
+
+    @classmethod
+    @retry(stop_max_attempt_number=5, wait_fixed=500)
+    def connect(cls):
+        cls.db_connection = psycopg2.connect(cls.db_connection_string)
+
+    def setUp(self):
+        setup_stub_aws_config
```

Comment:
Doesn't this need to be called? Otherwise, what is this doing?

Reply:
I believe `setUp()` and `tearDown()` are automatically called before and after each test respectively.

Reply:
I meant `setup_stub_aws_config` - as in, shouldn't it be `setup_stub_aws_config()`?